### PR TITLE
implement SDL-native sound looping for moving walls

### PR DESCRIPTION
**Pull request by fabiangreffrath**
@rfomin @mikeday0 @NightFright2k19
Could you guys check if this fixes the clicking sound issues for you?

### CODE REVIEW

**Comment by mikeday0** [comment]
Only had a little bit of time for testing but the clicking seems to be gone for me! I checked on Going Down map 11 and DBP25 map 01.

**Comment by rfomin** [comment]
I tested the DBP25 and the wads from this issue: [fabiangrefrath/crispy-doom#794](https://github.com/fabiangreffrath/crispy-doom/issues/794) and everything works great! Thank you.

**Comment by NightFright2k19** [comment]
I don't remember any pwad right now where I encountered the issue recently. I know there was a map that was truly horrible with lots of overlaying identical sounds, but it was probably in Project Einherjar or Operation Midgard, which are GZDoom mods,

**Review by rfomin — APPROVED** [review]

**Comment by fabiangreffrath** [comment]
There's still some edge cases I need to handle, probably later today.

**Comment by fabiangreffrath** [comment]
Short checklist:

- [ ] Looping sounds must stop when entering the menu (in singleplayer mode)
- [ ] Looping sounds must stop when entering pause
- [ ] Looping sounds must stop when exiting the level
- [ ] Looping sounds must stop when dying (in singleplayer mode)

In other words, if the textures keep visibly moving, the looping sounds must continue. If the textures stop moving, the looping sounds must stop as well.

File: src/p_floor.c [hunk 1]
```diff
@@ -254,7 +254,7 @@ void T_MoveFloor(floormove_t* floor)
   );
   
   if (!(leveltime&7))     // make the floormove sound
-    S_StartSound((mobj_t *)&floor->sector->soundorg, sfx_stnmov);
```
Comment (kraflab) [inline src/p_floor.c:257]: This flow works if there is a final sound that the soundorg makes when the loop should stop, right? I can see that for instance woof's floors do always make a sound - when they stop naturally. Is there anything that can interrupt a moving plane without making a sound? (In which case, the looping sound would continue in perpetuity.)

File: src/p_ceilng.c [hunk 2]
```diff
@@ -84,7 +84,7 @@ void T_MoveCeiling (ceiling_t* ceiling)
           case genSilentCrusher:
             break;
           default:
-            S_StartSound((mobj_t *)&ceiling->sector->soundorg,sfx_stnmov);
```
Comment (kraflab) [inline src/p_ceilng.c:87]: Where is this looping sound terminated in a general ceiling raise case?

File: src/p_floor.c (hunk 1 above)
Comment (kraflab) [reply src/p_floor.c:257]: Here's an example of the type of edge case that concerns me 😅 
```c
          case 163:
            // Stop Moving floor
            // 163 S1  EV_DoPlat(perpetualRaise,0)
            EV_StopPlat(line);
            P_ChangeSwitchTexture(line,0);
            break;
```

**Pull request merged**